FIAT/macro.py

```diff
@@ -360,11 +360,12 @@ class CkPolynomialSet(polynomial_set.PolynomialSet):
     :arg ref_el: The simplicial complex.
     :arg degree: The polynomial degree.
     :kwarg order: The order of continuity across subcells.
+    :kwarg vorder: The order of super-smoothness at interior vertices.
     :kwarg shape: The value shape.
     :kwarg variant: The variant for the underlying ExpansionSet.
     :kwarg scale: The scale for the underlying ExpansionSet.
     """
-    def __init__(self, ref_el, degree, order=1, shape=(), **kwargs):
+    def __init__(self, ref_el, degree, order=1, vorder=0, shape=(), **kwargs):
```

Comment:
Is supersmoothness a condition we impose or something that emerges organically as a result of all the other conditions as a consequence?  (No harm if we impose extra conditions since we're just taking a null space -- unless we hit numerical precision issues).

Reply:
It's an extra condition. I first compute a basis that is C^k across codimension 1 facets with an SVD, this should automatically give C^{k+1} supersmoothness at the vertex. To prevent different scales in the singular values, I then impose the supersmoothness at the interior vertices one order at a time by computing separate SVDs.